pkg/sdk/secrets_def.go

```diff
@@ -2,26 +2,50 @@ package sdk
 
 import (
 	"fmt"
+	"strings"
 
 	g "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/poc/generator"
 )
 
 //go:generate go run ./poc/main.go
 
-type OauthSecretType string
+type (
+	SecretType string
+)
 
-func NewOauthSecretType(s OauthSecretType) *OauthSecretType {
-	return &s
+func ToSecretType(s string) (SecretType, error) {
+	switch strings.ToUpper(s) {
+	case string(SecretTypePassword):
+		return SecretTypePassword, nil
+	case string(SecretTypeOAuth2):
+		return SecretTypeOAuth2, nil
+	case string(SecretTypeGenericString):
+		return SecretTypeGenericString, nil
+	case string(SecretTypeOAuth2ClientCredentials):
+		return SecretTypeOAuth2ClientCredentials, nil
+	case string(SecretTypeOAuth2AuthorizationCodeGrant):
+		return SecretTypeOAuth2AuthorizationCodeGrant, nil
+	default:
+		return "", fmt.Errorf("invalid secret type: %s", s)
+	}
 }
 
 const (
-	SecretTypePassword                               = "PASSWORD"
-	SecretTypeOAuth2                                 = "OAUTH2"
-	SecretTypeGenericString                          = "GENERIC_STRING"
-	OAuth2ClientCredentialsFlow      OauthSecretType = "CLIENT_CREDENTIALS"       // #nosec G101
-	OAuth2AuthorizationCodeGrantFlow OauthSecretType = "AUTHORIZATION_CODE_GRANT" // #nosec G101
+	SecretTypePassword                     SecretType = "PASSWORD"
```

Comment:
nit: I don't like that this is one enum here. Two different use cases are mixed here: SQL TYPE value and creation variants. I'm okay with merging this but let's talk later how we can improve this.